drivers/pwm/pwm_led_esp32.c

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright (c) 2017 Vitor Massaru Iha <vitor@massaru.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Include esp-idf headers first to avoid redefining BIT() macro */
+#include <esp_intr_alloc.h>
+#include <soc/dport_reg.h>
+#include <rom/gpio.h>
+#include <soc/gpio_sig_map.h>
+#include <soc/ledc_reg.h>
+
+#include <soc.h>
+#include <errno.h>
+#include <board.h>
+#include <pwm.h>
+#include <kernel.h>
+#include <gpio.h>
+#include <string.h>
+
+enum {
+	PWM_LED_ESP32_REF_TICK_FREQ,
+	PWM_LED_ESP32_APB_CLK_FREQ,
+};
+
+enum {
+	PWM_LED_ESP32_HIGH_SPEED,
+	PWM_LED_ESP32_LOW_SPEED
+};
+
+#define PWM_LED_ESP32_CHANNEL_NOT_USED -1
+
+/* XXX Move struct peripheral for other place ? */
```

Comment:
Yeah, I agree. This should be a SoC-specific thing; writing to `DPORT_PERIP_*_REG` happens in other ESP32 drivers.